### PR TITLE
Upgrade Docker for Linux

### PR DESCRIPTION
**Pull request by mumumumu**
There's a new flag in a later version of docker I'd like to make use of. 

Also went ahead and bumped the docker compose version.

### CODE REVIEW

File: packer/windows/scripts/install-docker.ps1 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Stop script execution when a non-terminating error occurs
 $ErrorActionPreference = "Stop"
 
-$docker_version="20.10.9"
+$docker_version="20.10.23"
```
Comment (triarius) [inline packer/windows/scripts/install-docker.ps1:4]: Looks like the latest version from the DockerMsftProvider is still
20.10.9. The build logs have:
```
[0;32m2023-02-22T01:10:00Z:     amazon-ebs: Upgrading DockerMsftProvider module[0m

[0;32m2023-02-22T01:10:09Z:     amazon-ebs: Printing available docker versions[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs:[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Name                           Version          Source           Summary[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: ----                           -------          ------           -------[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         17.06.2-ee-25    DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         18.03.1-ee-12    DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         18.09.14         DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         19.03.19         DockerDefault    Contains Docker EE for use with Windows Server.[0m

[0;32m2023-02-22T01:10:30Z:     amazon-ebs: Docker                         20.10.9          DockerDefault    Contains Docker EE for use with Windows Server.[0m
```

**Review by triarius — COMMENTED** [review]
Thanks for making this PR @mumumumu. It looks like the current means of installing Docker EE on Windows only go up to 20.10.9, so if you remove that commit, we can get this merged.

We'll have a look at how to support more recent Docker EE versions on Windows Server separately. If you have thoughts, we would gladly welcome another PR.

File: packer/windows/scripts/install-docker.ps1 (hunk 1 above)
Comment (triarius) [reply packer/windows/scripts/install-docker.ps1:4]: This is the cause of the build failure on [`f87c3da` (#1093)](https://github.com/buildkite/elastic-ci-stack-for-aws/pull/1093/commits/f87c3da08f5a312d2649d127c356c9feb445c406)

**Comment by mumumumu** [comment]
@triarius thanks for taking a look, I've un-bumped the Windows version.

**Review by triarius — APPROVED** [review]
LGTM

**Pull request merged**